kotest-assertions/kotest-assertions-json/src/jvmMain/kotlin/io/kotest/assertions/json/JsonMatchers.jvm.kt

```diff
@@ -0,0 +1,85 @@
+package io.kotest.assertions.json
+
+import java.io.File
+import java.nio.file.Path
+import kotlin.io.path.readText
+
+fun Path.shouldBeEmptyJsonArray(): Path {
+   this.readText().shouldBeEmptyJsonArray()
+   return this
+}
+
+fun File.shouldBeEmptyJsonArray(): File {
+   this.readText().shouldBeEmptyJsonArray()
+   return this
+}
+
+fun Path.shouldBeEmptyJsonObject(): Path {
+   this.readText().shouldBeEmptyJsonObject()
+   return this
+}
+
+fun File.shouldBeEmptyJsonObject(): File {
+   this.readText().shouldBeEmptyJsonObject()
+   return this
+}
+
+fun Path.shouldBeJsonArray(): Path {
+   this.readText().shouldBeJsonArray()
+   return this
+}
+
+fun File.shouldBeJsonArray(): File {
+   this.readText().shouldBeJsonArray()
+   return this
+}
+
+fun Path.shouldNotBeJsonArray(): Path {
+   this.readText().shouldNotBeJsonArray()
+   return this
+}
+
+fun File.shouldNotBeJsonArray(): File {
+   this.readText().shouldNotBeJsonArray()
+   return this
+}
+
+fun Path.shouldBeJsonObject(): Path {
+   this.readText().shouldBeJsonObject()
+   return this
+}
+
+fun File.shouldBeJsonObject(): File {
+   this.readText().shouldBeJsonObject()
+   return this
+}
+
+fun Path.shouldNotBeJsonObject(): Path {
+   this.readText().shouldNotBeJsonObject()
+   return this
+}
+
+fun File.shouldNotBeJsonObject(): File {
+   this.readText().shouldNotBeJsonObject()
+   return this
+}
+
+fun Path.shouldBeValidJson(): Path {
+   this.readText().shouldBeValidJson()
+   return this
+}
+
+fun File.shouldBeValidJson(): File {
+   this.readText().shouldBeValidJson()
+   return this
+}
+
+fun Path.shouldNotBeValidJson(): Path {
+   this.readText().shouldNotBeValidJson()
+   return this
+}
+
```

Comment:
can we add unit tests for all new code?

Reply:
Does it need unit tests?
because it's just the exact same as existing methods, and there are many methods which have similar patterns (ie. just light wrappers over existing methods which already have tests) which don't have unit tests

but if desired, I can add unit tests for all of those methods

Reply:
> Does it need unit tests? because it's just the exact same as existing methods, and there are many methods which have similar patterns (ie. just light wrappers over existing methods which already have tests) which don't have unit tests
> 
> but if desired, I can add unit tests for all of those methods

agreed, some such wrappers don't have tests at this time, but they should. and at least some of the new functions in this PR should have unit tests as well.

Reply:
yeah I'll write tests for them, I was just unsure of the test policy as similar functions didn't have tests

Reply:
I've copied over all the tests from the common tests, with the following changes:

- use `FunSpec` everywhere for consistency
- code reformat
- run tests against `File`s and `Path`s